pints/_mcmc/_differential_evolution.py

```diff
@@ -60,33 +67,65 @@ def __init__(self, chains, x0, sigma0=None):
         # Gamma
         self._gamma = 2.38 / np.sqrt(2 * self._dimension)
 
-        # Normal proposal std.
-        self._b = 0.01
+        # Gamma switch to 1 every (below) steps to help find
+        # modes
+        self._gamma_switch_rate = 10
+
+        # Error scale width
+        self._b = 0.001
+
+        # Normal error vs uniform
+        self._normal_error = True
+
+        # Relative scaling
+        self._relative_scaling = True
 
     def ask(self):
         """ See :meth:`pints.MultiChainMCMC.ask()`. """
         # Initialise on first call
         if not self._running:
             self._initialise()
 
+        # set gamma to 1
+        if self._iter_count % self._gamma_switch_rate == 0:
+            self._gamma = 1
+        self._iter_count += 1
+
         # Propose new points
         if self._proposed is None:
             self._proposed = np.zeros(self._current.shape)
             for j in range(self._chains):
+                if self._normal_error:
+                    error = np.random.normal(0, self._b_star, self._mu.shape)
+                else:
+                    error = np.random.uniform(-self._b_star, self._b_star,
+                                              self._mu.shape)
                 r1, r2 = r_draw(j, self._chains)
                 self._proposed[j] = (
                     self._current[j]
                     + self._gamma * (self._current[r1] - self._current[r2])
-                    + np.random.normal(
-                        0, np.abs(self._b * self._mu), self._mu.shape)
+                    + error
                 )
 
+            # reset gamma
+            self._gamma = 2.38 / np.sqrt(2 * self._dimension)
+
             # Set as read only
             self._proposed.setflags(write=False)
 
         # Return proposed points
         return self._proposed
 
+    def set_normal_error(self, normal_error):
+        """
+        If true sets the error process to be a normal
```

Comment:
Please break lines at 79 chars! (You can probably set up your editor to display a gray line or something)